packages/cli-repl/src/mongosh-repl.spec.ts

```diff
@@ -566,19 +566,19 @@ describe('MongoshNodeRepl', () => {
     it('colorizes input statement', async() => {
       input.write('const cat = "Nori"');
       await tick();
-      expect(output).to.match(/const(\x1b\[.*m)+ cat = (\x1b\[.*m)+"Nori"(\x1b\[.*m)+/);
+      expect(output).to.match(/const(\x1b\[.*m)+ cat = (\x1b\[.*m)+"(\x1b\[.*m)+N(\x1b\[.*m)+o(\x1b\[.*m)+r(\x1b\[.*m)+i(\x1b\[.*m)+"(\x1b\[.*m)+/);
     });
 
     it('colorizes input function', async() => {
       input.write('function add (a, b) { return a + b }');
       await tick();
-      expect(output).to.match(/function(\x1b\[.*m)+ (\x1b\[.*m)+add(\x1b\[.*m)+ \(a, b\) \{ (\x1b\[.*m)+return(\x1b\[.*m)+ a \+ b/);
+      expect(output).to.match(/function(\x1b\[.*m)+ (\x1b\[.*m)+a(\x1b\[.*m)+d(\x1b\[.*m)+d(\x1b\[.*m)+ \(a, b\) \{ retur\x08+(\x1b\[.*m)+return(\x1b\[.*m)+ a \+ b/);
```

Comment:
```suggestion
      expect(output).to.match(/function(\x1b\[.*m)+ (\x1b\[.*m)+a(\x1b\[.*m)+d(\x1b\[.*m)+d(\x1b\[.*m)+ \(a, b\) \{\x08+(\x1b\[.*m)+return(\x1b\[.*m)+ a \+ b/);
```
Some returns are mangled in there 🤔

Reply:
Oh, this is on purpose – You enter `retur`, which is not a keyword, then you enter `n`, which *makes* it a keyword, so we go back a few times (\x08 is backspace), and overwrite it with the highlighted version :)

Reply:
Also, to be clear, this is the same visual behavior as before – the difference is that we can’t match the complete and fully highlighted string anymore, because we’re not printing the full line, only applying diffs to it as the user input is being entered

Reply:
🤯